src/jsxc.lib.gui.js

```diff
@@ -2644,20 +2644,57 @@ jsxc.gui.window = {
          jsxc.xmpp.sendMessage(message);
       }
 
-      jsxc.gui.window._postMessage(message);
+      /*
+       New Line added to sort the messages
+      */
+      jsxc.gui.window.sortMessages(message);
 
       if (message.direction === 'out' && message.msg === '?' && jsxc.options.get('theAnswerToAnything') !== false) {
          if (typeof jsxc.options.get('theAnswerToAnything') === 'undefined' || (Math.random() * 100 % 42) < 1) {
-            jsxc.options.set('theAnswerToAnything', true);
+	         jsxc.options.set('theAnswerToAnything', true);
 
-            jsxc.gui.window.postMessage(new jsxc.Message({
-               bid: message.bid,
-               direction: jsxc.Message.SYS,
-               msg: '42'
-            }));
+	         jsxc.gui.window.postMessage(new jsxc.Message({
+	            bid: message.bid,
+	            direction: jsxc.Message.SYS,
+	            msg: '42'
+	         }));
          }
       }
+      return message;
+   },
+
+   sortMessages: function(message){
+      var messageArray = jsxc.options.allMessages;
```

Comment:
I was trying to do it in some other ways. For that, I initialized an empty data structure in a function. So whenever the function ran it again got initialized to empty.

For example: say I am declaring a variable in sortMessage function.
`sortMessages: function(message) {
    var messageArray = []
}`
So whenever the function ran it again got initialized to empty deleting all the messages. 

If you have any other idea please share.